src/ranking.jl

```diff
@@ -171,3 +171,108 @@ Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; sortkwargs...) =
     _rank(_tiedrank!, x, Float64; sortkwargs...)
+
+"""
+    quantilerank(a::RealVector, score)
+    quantilerank(a::RealVector, score::RealVector)
+
+Return the percentile-position of score (0-1) relative to a.
+
+For example, a quantilerank of 80% means that 80% of the values in `a` are below the given score.
+
+# Arguments
+* `a`: Array of values to which `score` is compared.
+* `score`: Value that is compared to the elements in `a`.
+* `method` (optional, default `:rank`):  method used for calculate the quantilerank. Options available (`:rank`, `:weak`, `:strict`, `:mean`).
+    1. `:rank` - Average percentage ranking of score. In case of multiple matches, average the percentage rankings of all matching scores.
+    2. `:weak` - This method corresponds to the definition of a cumulative distribution function. A quantilerank of 80% means that 80% of values are less than or equal to the provided score.
+    3. `:strict` - Similar to :weak, except that only values that are strictly less than the given score are counted.
+    4. `:mean` - The average of the :weak and :strict scores, often used in testing. See https://en.wikipedia.org/wiki/Percentile_rank
```

Comment:
I tested the code using the examples from the excel and libreoffice documentation and none of the methods got the same result, confirming that there is no equivalence. So far,the current equivalence is that of scipy. 

I have been looking up bibliographic references for `percentilerank`and it has been difficult to find and this worries me.
In [Excel ](https://support.microsoft.com/en-us/office/percentrank-inc-function-149592c9-00c0-49ba-86c1-c1f45b80463a) there are no references or source code. In [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.percentileofscore.html), only source code, no reference. In [LibreOffice](http://docs.oasis-open.org/office/v1.2/os/OpenDocument-v1.2-os-part2.html#__RefHeading__1018738_715980110) there is a brief description of the method used, but also without references. In [dplyr (R)](https://dplyr.tidyverse.org/reference/ranking.html), they emulate the percent_rank of [SQL2003](https://docs.microsoft.com/en-us/sql/t-sql/functions/percent-rank-transact-sql?view=sql-server-ver15), whose documentation has no source code, just description and example. On Wikipedia, there's only the reference to the book I've already cited.

If after all my comments on this first review, you find that it's not worth implementing the function, please know that I have no problem closing this PR

Reply:
> It would be nice to implement these definitions in addition to those from NumPy. In particular, given that quantile is equivalent to PERCENTILE/PERCENTILE.INC in Excel, which is the perfect inverse of PERCENTRANK/PERCENTRANK.INC in Excel (in my tests), it would make sense to use that definition by default for consistency.

For mantain the consistency you suggested I changed the name of the function from `percentilerank` to `percentrank`. I coded the "INC" and "EXC" method as argument of the functions. If you prefer, I could create another function called percentrank.inc and percentrank.exc. Whatever you say.


I also checked for all those implementations links you send me and implemented them (SQL (mariadb), EXC and INC from LibreOffice/Excel). 
Also implemented the interpolation presented in the libreoffice source code.